### PR TITLE
TASK: Improve performance on content cache flush

### PR DESCRIPTION
**Pull request by dlubitz**
To improve the performance on content cache flush, we removed one of the two anchor point for cache flushing. So we take the risk of a slightly outdated cache in some cases, but we will not perform the cache flush twice and reduce the workload.

See for more information: https://github.com/neos/neos-development-collection/blob/9.0/Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php#L44-L113

### CODE REVIEW

File: Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php [hunk 1]
```diff
@@ -110,28 +110,22 @@
  * => INVARIANT_2 violated.
  * => this case needs a cache flush at onBeforeBatchCompleted.
  *
- * SUMMARY: we need to flush the cache at BOTH places.
+ * SUMMARY: We need to flush the cache at BOTH places to ensure non of the cases above happens. BUT (!)
+ *          Flushing the cache is a very expensive operation, so we need to reduce them as much as we can.
+ *          So we decided to take the risk of CASE B, which might show an outdated content a bit longer than
+ *          it might be expected.
```
Comment (bwaidelich) [inline Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php]: Just wondering: Who is "we" in this case?

Comment (dlubitz) [reply Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php]: The responsible "Cache Flushing Team of the Day". We rotate and everyone has to do their duty.

Comment (dlubitz) [reply Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php]: 😝

Comment (bwaidelich) [reply Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php]: tehe.. I didn't ask to disrespect anyone. I'm just missing some context as to _why_ this was decided this way (and, frankly, if I was part of it 😀 )

Comment (dlubitz) [reply Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php]: Ahhh... you referring to the second "we" (decided to take the risk) ... 🙈 
I thought you and me discussed that, while we looked into the overhauling of the content cache flusher. But maybe I'm wrong and it was @kitsunet ... 🤔

Comment (kitsunet) [reply Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php]: Yep that was me I think. :)

**Review by kitsunet — APPROVED** [review]

**Comment by dlubitz** [comment]
I've updated the documentation with @skurfuerst.

**Review by skurfuerst — APPROVED** [review]
synchronously reviewed with @dlubitz  - IMHO is safe as we run synchronously now :)

**Pull request merged**